docs/01-app/01-getting-started/06-partial-prerendering.mdx

```diff
@@ -230,7 +230,7 @@ import { Suspense } from 'react'
 export default function Page({
   searchParams,
 }: {
-  searchParams: Promise<{ sort: string }>
+  searchParams: Promise<{ sort?: string }>
```

Comment:
Can they also be `undefined`? There's a distinction between a value missing and a value being `undefined`. TypeScript may check this with [exactOptionalProperties](https://www.typescriptlang.org/tsconfig/#exactOptionalPropertyTypes)

Reply:
I don't think Next.js will ever pass a search parameter into a page as `undefined`, but I'm not completely sure. With this PR, the following works (is valid):

```ts
  searchParams: Promise<{ tag?: string }>
```

But this throws a TypeScript error:

```ts
  searchParams: Promise<{ tag: string | undefined }>
```

Is this behavior OK?